### PR TITLE
Stanford Diagrams

### PR DESCRIPTION
**Pull request by FreezinG117**
Add support for [Stanford Diagrams](https://gssc.esa.int/navipedia/index.php/The_Stanford_%E2%80%93_ESA_Integrity_Diagram:_Focusing_on_SBAS_Integrity).

### CODE REVIEW

**Comment by kt3k** [comment]
Wow, this is an amazing patch 🤯

I opened `chart_stanford.html` and `chart_stanford_custom_elements.html`, and both show the error:
```
Error: number of elements necessary for Stanford graph type is invalid, must be: json {somekey, somekey, epochs}
```
Can you check these examples?

I will review soon in more details.

**Comment by FreezinG117** [comment]
Ups, it should work now 🙂

File: spec/stanford-spec.js [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+        it('should return 100% if the region has all the epochs', function () {
+            var region = [
+                {x: 0, y: 0},
+                {x: 60, y: 0},
+                {x: 60, y: 60},
+                {x: 0, y: 60}
+            ];
+
+            var result = chart.internal.countEpochsInRegion(region);
```
Comment (kt3k) [inline spec/stanford-spec.js]: Nice unit test 👍

File: src/type.js [hunk 2]
```diff
@@ -55,7 +55,11 @@ ChartInternal.prototype.isBarType = function (d) {
 };
 ChartInternal.prototype.isScatterType = function (d) {
     var id = isString(d) ? d : d.id;
-    return this.config.data_types[id] === 'scatter';
+    return this.config.data_types[id] === 'scatter' || this.config.data_types[id] === 'stanford';
```
Comment (kt3k) [inline src/type.js]: I think this is confusing. How about creating another method `isScatterOrStanfordType` and replace the usages of `isScatterType` with it where necessary.

**Comment by kt3k** [comment]
I commented one point. Please check.

This has nice tests, nice documentation and examples. The implementation is enough independent of other part of source code. I think it's reasonable to include this chart to c3.

Thank you for the great contribution!

File: src/type.js (hunk 2 above)
Comment (FreezinG117) [reply src/type.js]: @kt3k I did what you suggested, see if the changes I made work for you. Thanks!

**Comment by kt3k** [comment]
released as v0.7.0. Thank you for the huge contribution! https://c3js.org/samples/chart_stanford.html

**Pull request merged**